v4.0.0/tasks-schema.json

```diff
@@ -0,0 +1,1496 @@
+                                    "start",
+                                    "end"
+                                ]
+                            }
+                        ],
+                        "required": [
+                            "start",
+                            "end"
+                        ]
+                    },
+                    "last_data_date": {
+                        "description": "The last date with recorded data in the data set used as input to a model.",
+                        "examples": [
+                            "2022-07-18"
+                        ],
+                        "type": "string",
+                        "format": "date"
+                    },
+                    "file_format": {
+                        "description": "Accepted file formats of model output files for the round. Overrides the file formats provided in admin.json.",
+                        "examples": [
+                            [
+                                "arrow",
+                                "parquet"
+                            ],
+                            [
+                                "csv"
+                            ]
+                        ],
+                        "type": "array",
+                        "items": {
+                            "type": "string",
+                            "enum": [
+                                "csv",
+                                "parquet",
+                                "arrow"
+                            ]
+                        }
+                    }
+                },
+                "required": [
+                    "round_id_from_variable",
+                    "round_id",
+                    "model_tasks",
+                    "submissions_due"
+                ],
+                "additionalProperties": true
+            }
+        },
+        "output_type_id_datatype": {
+            "description": "The hub level data type of the output_type_id column. This data type must be shared across all files in the hub and be able to represent all output type ID values across all hub output types and rounds. If not provided or set to 'auto', hub defaults to autodetecting the simplest hub level data type.",
+            "default": "auto",
+            "examples": [
+                "character"
+            ],
+            "type": "string",
+            "enum": [
+                "auto",
+                "character",
+                "double",
+                "integer",
+                "logical",
+                "Date"
+            ]
+        }
+    },
+    "derived_task_ids": {
+        "description": "Names of derived task IDs, i.e. task IDs whose values are derived from (and therefore dependent on) the values of other variables.",
+        "type": [
+            "array",
+            "null"
+        ],
+        "uniqueItems": true,
+        "items": {
+            "type": "string"
+        }
+    },
```

Comment:
I feel like it's a hub property overall in that I can't see many situations where a task ID is derived in one round and not derived in another. The chances of it being stable are far greater than the property changing by round so it would be annoying to have to re-define it again in every round. 

I'll make sure however that if a new derived task ID is added to a new round and the task id name added to `derived_task_ids`, it will not affect validation of older rounds, i.e. if the task ID is not present in the data, `derived_task_ids` will just be ignored.

The is the potential to allow for a  `derived_task_ids` property at the  round level that overrides the overall property but I think we can wait and see if anyone requests that as it feels like an extreme edge case?

Reply:
Thank you for the additional information. 

You make a really good point, however I am worrying a bit about the scenario hubs. They tend to have lot of rounds and have different tasks id for some round as for the need of a specific round or scenario you might need to create new column that are not replicated in the next round. 
I am not sure how frequent this new columns will be tagged as derived tasks ids, but even saying that it looks to me a little bit weird to have the tasks id as hub level as they are not define at hub level but at round level.

Reply:
I definitely don't want everyone to have to respecify this property at each round as for the majority of hubs this is very stable so we will for sure keep the overall specification. As mentioned there is the option for overriding at the round but I do not want to implement unless its actually necessary

Reply:
> The problem though does not arise from a derived task id not being in all rounds 

I misunderstood and I was afraid that would be an issue but if not, that solve some of my problems! Apologize for that! 
I agree that changing the behavior of a column is bad practice and should not be supported.

> I definitely don't want everyone to have to respecify this property at each round as for the majority of hubs this is very stable so we will for sure keep the overall high level specification.

I also agree, that's to much to ask for something that might not be used often. 
And the optional overriding seems like a good idea,  and I think a hub can do their own wrapper/function to deal with it if necessary.

So to summarize, thank you for the additional information, I change my mind and I am ok with it being at hub level!